test/e2e/kubernetes/kubernetes.go

```diff
@@ -251,3 +257,99 @@ func EnsureNodeWithIPIsDeleted(ctx context.Context, k8s *kubernetes.Clientset, i
 	}
 	return nil
 }
+
+func WaitForHybridNodeToUpgrade(ctx context.Context, k8s *kubernetes.Clientset, nodeName, targetVersion string, logger logr.Logger) (*corev1.Node, error) {
+	foundNode := &corev1.Node{}
+	consecutiveErrors := 0
+	err := wait.PollUntilContextTimeout(ctx, nodePodDelayInterval, hybridNodeUpgradeTimeout, true, func(ctx context.Context) (done bool, err error) {
+		node, err := k8s.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+		if err != nil {
+			consecutiveErrors += 1
+			logger.Info("consecutiveErrors", "consecutiveErrors", consecutiveErrors)
+			if consecutiveErrors > 3 {
+				return false, fmt.Errorf("getting hybrid node %s: %w", nodeName, err)
+			}
+			logger.Info("Retryable error getting hybrid node. Continuing to poll", "name", nodeName, "error", err)
+			return false, nil // continue polling
+		}
+		consecutiveErrors = 0
+
+		kubernetesVersion := strings.TrimPrefix(node.Status.NodeInfo.KubeletVersion, "v")
+		// If the current version matches the target version of kubelet, return true to stop polling
+		if strings.HasPrefix(kubernetesVersion, targetVersion) {
+			foundNode = node
+			logger.Info("Node successfully upgraded to desired kubernetes version", "version", targetVersion)
+			return true, nil
+		}
+
+		return false, nil // continue polling
+	})
+	if err != nil {
+		return nil, fmt.Errorf("waiting for node %s kubernetes version to be upgraded to %s: %w", nodeName, targetVersion, err)
+	}
+
+	return foundNode, nil
+}
+
+func GetPreviousK8sVersion(kubernetesVersion string) (string, error) {
```

Comment:
Avoid using `get` unless you are actually calling a get API. This is not a getter but same argument applies imo: https://go.dev/doc/effective_go#Getters

```suggestion
func PreviousVersion(version string) (string, error) {
```